src/compiler/checker.ts

```diff
@@ -27266,11 +27293,36 @@ namespace ts {
             node: PropertyAccessExpression | QualifiedName | PropertyAccessExpression | VariableDeclaration | ParameterDeclaration | ImportTypeNode | PropertyAssignment | ShorthandPropertyAssignment | BindingElement,
             isSuper: boolean, writing: boolean, type: Type, prop: Symbol, reportError = true): boolean {
 
-            const flags = getDeclarationModifierFlagsFromSymbol(prop, writing);
             const errorNode = node.kind === SyntaxKind.QualifiedName ? node.right :
                 node.kind === SyntaxKind.ImportType ? node :
                 node.kind === SyntaxKind.BindingElement && node.propertyName ? node.propertyName : node.name;
 
+            const typeChecks = checkPropertyAccessibilityAtNode(node, isSuper, writing, type, prop);
+            switch (typeChecks.result) {
+                case (TypeChecks.Ok):
+                    return true;
+                case (TypeChecks.HasDiagnostic):
+                    if (reportError) {
+                        error(errorNode, typeChecks.message, ...typeChecks.args);
+                    }
+                    return false;
+            }
+        }
+
+        // Possible concerns:
+        // (1) I'm not sure if this function has the right behavior if `node` is allowed to be any node,
+        // although we only use `node` for its location in the parse tree.
+        // (2) Does it even make sense to check for property accessibility at a certain arbitrary node?

```

Comment:
yes, this is a completely sensible concept, and I think this is the right name for it. (or maybe "atLocation", but that's usually used by functions used directly by the API or language service)